deepchecks/base/dataset.py

```diff
@@ -783,7 +783,7 @@ def ensure_not_empty_dataset(cls, obj: t.Any, *, cast: bool = False) -> 'Dataset
                 )
 
     @classmethod
-    def datasets_share_features(cls, datasets: t.List['Dataset']) -> bool:
+    def datasets_share_features(cls, *datasets: t.List['Dataset']) -> bool:
```

Comment:
change `datasets` type from `List[Dataset]` to `Dataset`, currently it means `Iterable[List[Dataset]]`